File: src/main/java/fr/aiko/Karl/parser/ast/values/NullValue.java
package fr.aiko.Karl.parser.ast.values;

import fr.aiko.Karl.parser.TokenType;

public class NullValue extends Value {
    public final String value;

    public NullValue(String value) {
        this.value = value;
    }

    @Override
    public String toString() {
        return value;
    }

    @Override
    public int toInt() {
        return 0;
    }

    @Override
    public float toFloat() {
        return 0;
    }

    @Override
    public TokenType getType() {
        return TokenType.NULL;
    }
}

File: src/main/kotlin/studio/karllang/karl/parser/ast/values/NullValue.kt
package studio.karllang.karl.parser.ast.values

import studio.karllang.karl.lexer.TokenType

class NullValue(val value: String) : Value() {
    override val type: TokenType = TokenType.NULL

    override fun toString(): String {
        return value
    }

    override fun toInt(): Int {
        return 0
    }

    override fun toFloat(): Float {
        return 0f
    }
}